src/Adapters/NLogTarget.Shared/ApplicationInsightsTarget.cs

```diff
@@ -1,202 +1,212 @@
+            }
+            else
+            {
+                this.SendTrace(logEvent);
+            }
+        }
+
+        private static string GetAssemblyVersion()
+        {
+            return typeof(ApplicationInsightsTarget).Assembly.GetCustomAttributes(false)
+                    .OfType<AssemblyFileVersionAttribute>()
+                    .First()
+                    .Version;
+        }
+
+        private void SendException(LogEventInfo logEvent)
+        {
+            var exceptionTelemetry = new ExceptionTelemetry(logEvent.Exception)
+            {
+                SeverityLevel = this.GetSeverityLevel(logEvent.Level)
+            };
+
+            this.BuildPropertyBag(logEvent, exceptionTelemetry);
+            this.telemetryClient.Track(exceptionTelemetry);
+        }
+
+        private void SendTrace(LogEventInfo logEvent)
+        {
+            string logMessage = this.Layout.Render(logEvent);
+            
+            var trace = new TraceTelemetry(logMessage)
+            {
+                SeverityLevel = this.GetSeverityLevel(logEvent.Level)
+            };
+
+            this.BuildPropertyBag(logEvent, trace);
+            this.telemetryClient.Track(trace);
+        }
+
+        private void BuildPropertyBag(LogEventInfo logEvent, ITelemetry trace)
+        {
+            trace.Timestamp = logEvent.TimeStamp;
+            trace.Sequence = logEvent.SequenceID.ToString(CultureInfo.InvariantCulture);
+
+            IDictionary<string, string> propertyBag;
+
+            if (trace is ExceptionTelemetry)
+            {
+                propertyBag = ((ExceptionTelemetry)trace).Properties;
+            }
+            else
+            {
+                propertyBag = ((TraceTelemetry)trace).Properties;
+            }
+
+            if (!string.IsNullOrEmpty(logEvent.LoggerName))
+            {
+                propertyBag.Add("LoggerName", logEvent.LoggerName);
+            }
+
+            if (logEvent.UserStackFrame != null)
+            {
+                propertyBag.Add("UserStackFrame", logEvent.UserStackFrame.ToString());
+                propertyBag.Add("UserStackFrameNumber", logEvent.UserStackFrameNumber.ToString(CultureInfo.InvariantCulture));
+            }
+
+            var properties = logEvent.Properties;
+            if (properties != null)
+            {
+                foreach (var keyValuePair in properties)
+                {
+                    string key = keyValuePair.Key.ToString();
+                    object valueObj = keyValuePair.Value;
+                    if (valueObj == null)
+                    {
+                        continue;
+                    }
+
+                    string value = valueObj.ToString();
+                    if (propertyBag.ContainsKey(key))
+                    {
+                        if (value == propertyBag[key])
+                        {
+                            continue;
+                        }
+                        key += "_1";

```

Comment:
Just hardcoding "_1" won't deal with more than a single duplicate, what happens when "_1" is already present?